src/megatron/energon/flavors/webdataset/metadata.py

```diff
@@ -158,7 +159,12 @@ def get_dataset_type(path: EPath) -> EnergonDatasetType:
             return EnergonDatasetType.METADATASET
         else:
             return EnergonDatasetType.INVALID
-    elif check_dataset_info_present(path):
-        return EnergonDatasetType.WEBDATASET
+    elif path.is_dir():
```

Comment:
I believe `is_dir` is not optimal with some remote filesystems, it may be unefficient. I'd rather directly check for the files that we're looking for.